src/components/Slider/SliderInfix.jsx

```diff
@@ -0,0 +1,69 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { bem, INFIX_KIND } from "./SliderCommons";
+import { useSliderInfix, useSliderSelection } from "./SliderContext";
+import Label from "../Label/Label";
+import Icon from "../Icon/Icon";
+
+const defaultIconProps = {
+  clickable: false,
+  ignoreFocusStyle: true
+};
+
+function getCurrentLabel({ isPostfix, isRange, valueText }) {
+  if (!isRange) {
+    return valueText;
+  }
+  if (isPostfix) {
+    return valueText[1];
+  }
+  return valueText[0];
+}
+
+function useSliderInfixComponent(kind) {
+  const { prefix, postfix, indicateSelection } = useSliderInfix();
+  const { isRange, value, valueText } = useSliderSelection();
+  const infix = kind === SliderInfix.kinds.POSTFIX ? postfix : prefix;
+
+  const isPostfix = kind === SliderInfix.kinds.POSTFIX;
+  if (indicateSelection && (isPostfix || isRange)) {
+    const currentLabel = getCurrentLabel({ isPostfix, isRange, valueText });
+    return [true, "", <Label text={currentLabel} color={Label.colors.DARK} kind={Label.kinds.LINE} />];
+  }
+  if (typeof infix === "object" && infix.icon) {
+    const { icon, ...restIconProps } = infix;
+    const iconProps = { ...defaultIconProps, ...restIconProps };
+    return [true, "", <Icon icon={icon} {...iconProps} />];
+  }
+  if (typeof infix === "function") {
+    return [true, "", infix(value, valueText)];
+  }
+  if (typeof infix === "string") {
+    return [true, "txt", infix];
+  }
+  if (typeof infix === "undefined") {
+    return [false, "", null];
+  }
+  return [true, "", infix];
```

Comment:
I would return an object with named fields as return value because it would be more readable and eaiser to understand what is true means and what is "" means.

Reply:
As at previous topics we have here pros and cons of each approach.
- currently we just using it in same file, it pretty clear how it works
- hook itself much more readable and short
- useState use same pattern - in future in case of reuse it provide easy ability to provide names according to they needs.

In any case I don't think it critical. I can change if if you prefer.

Reply:
So I would prefer to change it to object structure in that case, still think it is much more readable and easier to understand what is each value in the returning object means.